jack-matchmaker.service

```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=auto-connect JACK ports matching given patterns
+
+[Service]
+EnvironmentFile=/etc/conf.d/jack-matchmaker
+ExecStart=/bin/sh -c "/usr/bin/jack-matchmaker $${PATTERN_FILE+-p \"$PATTERN_FILE\"} $${EXACT_MATCHING:+-e} $${CLIENT_NAME+-N \"$CLIENT_NAME\"} $${CONNECT_INTERVAL+-I $CONNECT_INTERVAL} $${MAX_ATTEMPTS+-m $MAX_ATTEMPTS} $${VERBOSITY+-v $VERBOSITY} $${PATTERNS+$PATTERNS}"
```

Comment:
Uh, that's quite ugly, but it works :) Better to use `/bin/bash` instead of `/bin/sh`, though, since the extended variable expansion syntax is Bash-specific and some systems might not have `/bin/sh` linked to `/bin/bash`.

Reply:
If you use outer single quotes and just use `$$PATTERNS` (undefined variables just expand to an empty string), you can simplify the line slightly:

    ExecStart=/bin/bash -c '/usr/bin/jack-matchmaker $${PATTERN_FILE+-p "$PATTERN_FILE"} $${EXACT_MATCHING:+-e} $${CLIENT_NAME+-N "$CLIENT_NAME"} $${CONNECT_INTERVAL+-I $CONNECT_INTERVAL} $${MAX_ATTEMPTS+-m $MAX_ATTEMPTS} $${VERBOSITY+-v $VERBOSITY} $$PATTERNS'

Reply:
Thanks, updated :)

Reply:
Uh using single quotes doesn't seem to work.

Reply:
Uh single quotes doesn't seem to work